internal/net/grpc/pool/pool.go

```diff
@@ -201,6 +203,10 @@ func (p *pool) connect(ctx context.Context) (c Conn, err error) {
 			log.Debugf("establishing same connection to %s", p.addr)
 			conn, err := p.dial(ctx, p.addr)
 			if err != nil {
+				failCnt++
+				if p.isIP && (p.Len() <= 2 || failCnt >= p.Len()/3) {
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blueviolet.svg?style=flat-square)
Why the if statement checks a condition that `failCnt >= p.Len()/3`? Where is the `3` coming from?
This is just a question.

Reply:
If the IP connection fails, that is, the target IP is invalid or missing, or if there is a network failure, you need to consider it. If there are less than two pool connections or 30% of the pool should immediately return an error.